### ISSUE TITLE
Copies file to temp dir, but does nothing after

### ISSUE THREAD

**Issue by austeregrim** [ekim1337/PlexComskip#5]
I'm on a linux box, everything looks right. I can run the script with the plex user and it runs fine, but when plex itself runs it, it just doesn't run comskip on it.

**Comment by billybooth** [comment]
Same here on Debian.  It actually does run comskip, but comskip fails when Plex runs it as a post-processing task.  When the plex user has a shell, there is no issue.  I don't have my old logs, but they were somewhat informative.  I just re-setup the server after a drive failure, so I'll let you know if I'm able to work it out this time around.

**Comment by keithah** [comment]
I'm having the same issue, where if I run it from cmdline works fine, but when Plex runs it, seems to ot work:

2017-07-20 01:01:03,838 [7a094d] PlexComskip got invoked from /opt/PlexComskip/PlexComskip.py
2017-07-20 01:01:03,841 [7a094d] Using session ID: 7a094d33-fd57-46d7-8b39-2915fbd4afbb
2017-07-20 01:01:03,841 [7a094d] Using temp dir: /tmp/comskip/7a094d33-fd57-46d7-8b39-2915fbd4afbb
2017-07-20 01:01:03,841 [7a094d] Using input file: /home/keith/Media/DVR/m92p/TV/.grab/eded11a7ffa119cd594699ba89037e2860ff3d71/Expedition Unknown (2015) - S02E10 - Africa's Gold Hoard.ts
2017-07-20 01:01:03,841 [7a094d] Copying file to work on it: /tmp/comskip/7a094d33-fd57-46d7-8b39-2915fbd4afbb/Expedition Unknown (2015) - S02E10 - Africa's Gold Hoard.ts
2017-07-20 01:01:07,829 [7a094d] [comskip] Command: ['/usr/local/bin/comskip', '--output', '/tmp/comskip/7a094d33-fd57-46d7-8b39-2915fbd4afbb', '--ini', '/opt/PlexComskip/comskip.ini', "/tmp/comskip/7a094d33-fd57-46d7-8b39-2915fbd4afbb/Expedition Unknown (2015) - S02E10 - Africa's Gold Hoard.ts"]
2017-07-20 01:01:09,303 [7a094d] Using EDL: /tmp/comskip/7a094d33-fd57-46d7-8b39-2915fbd4afbb/Expedition Unknown (2015) - S02E10 - Africa's Gold Hoard.edl
2017-07-20 01:01:09,303 [7a094d] Keeping segment from 0.0 to the end of the file...
2017-07-20 01:01:09,304 [7a094d] [ffmpeg] Command: ['/usr/bin/ffmpeg', '-i', "/tmp/comskip/7a094d33-fd57-46d7-8b39-2915fbd4afbb/Expedition Unknown (2015) - S02E10 - Africa's Gold Hoard.ts", '-ss', '0.0', '-c', 'copy', 'segment-0.ts']
2017-07-20 01:01:09,997 [7a094d] Going to concatenate 0 files from the segment list.
2017-07-20 01:01:09,997 [7a094d] [ffmpeg] Command: ['/usr/bin/ffmpeg', '-y', '-f', 'concat', '-i', '/tmp/comskip/7a094d33-fd57-46d7-8b39-2915fbd4afbb/segments.txt', '-c', 'copy', "/tmp/comskip/7a094d33-fd57-46d7-8b39-2915fbd4afbb/Expedition Unknown (2015) - S02E10 - Africa's Gold Hoard.ts"]
2017-07-20 01:01:10,044 [7a094d] Sanity checking our work...
2017-07-20 01:01:10,044 [7a094d] Output file size was too similar (doesn't look like we did much); we won't replace the original: 2.9GB -> 2.9GB
2017-07-20 01:01:10,889 [7a094d] Done processing!

Anyone figure this out?

**Comment by forsethc** [comment]
I'm pretty sure the `.grab` files are temporary files that the DVR uses as it's recording.  I've written some shell scripts to automate commercial removal and had to specifically exclude `.grab` files.

EDIT: I replied after reading @keithah's comment, it may not apply to the original issue...

**Comment by austeregrim** [comment]
It's not the grab files. It's clearly executing on a non "grab" file. 
It's something wrong with the python environment being executed from a script. 
It can be replicated by calling the python script from a bash script. 

Calling the script directly with a file, vs. calling the script with a variable name that is a file path string is treated differently, as far as I can tell with my limited experience. 

I've been able to create a list of files, modify the list into a script that calls the comskip script per file on each line, it works. Make a script find each file and run it via a variable that has the path doesn't work.

I've had a lot of trouble calling python scripts this way. But I'm just not experienced enough  to trouble shoot, let alone have the time to bother.

I made a script that make a script to run comskip script individually on each file. It's so bastardized now... I can't even deal.

**Comment by forsethc** [comment]
Yeah, I read a different comment and missed the fact that you were calling it directly from Plex.

In my shell scripts I had to make sure I put quotes around the file/show name _and_ make sure the title didn't contain quotes.  Maybe something along those lines?

**Comment by austeregrim** [comment]
I'm sure I've got everything quoted properly. I'm guessing it's a comskip issue, not really your script itself. Specifically a python execution problem in debian.

**Comment by billybooth** [comment]
Yes, it does have to be either a Comskip issue or an issue with the way that Comskip is executed in this context.  It may be something as simple as needing to set up the PATH when under the post-processing context on Debian-based distributions.  Plex recently switched from init.d to service on Debian, so I'll write a simple bash script to move the recorded file to a temp path, attempt to process with Comskip only, and move it back.  If that works without establishing a path, then the needle moves back to PlexComskip.

**Comment by billybooth** [comment]
@austeregrim, so you're saying that if you have plex post-processing call a script to execute PlexComskip on a specific file, that file is processed correctly?  Now that is interesting, because I'm also confident I'm quoting everything correctly (I'm calling a script that calls PlexComskip rather than than calling PlexComskip directly).  In the "handoff" script, I'm logging the incoming paths.  I feel like the real test will be to take my existing script and just call Comskip instead of PlexComskip to see if the same issue persists.  If it does, then I'll try setting up the path to ensure Comskip can find lib.  If it still fails, we'll need to take this up with the Comskip developer rather than here.

**Comment by keithah** [comment]
I am getting: Unable to find a suitable output format
from `ffmpeg`, but when I run it manually as the Plex user, it's fine.

Similar to: https://forums.plex.tv/discussion/283106/linux-post-script-not-working-from-plex-but-works-manually#latest

I gave up and sudo'd the Plexcomskip.py process in my script and it works now.. so could be on the Plex side possibly? So strange though.

**Comment by AiRC-ai** [comment]
how did you sudo the script to run in plex? 
i am still having the issue of not being able to run from plex but can from user cli and plex cli

**Comment by AiRC-ai** [comment]
this is the python code i believe i am having trouble with

    if TRANSCODE:
      output_file = os.path.join(temp_dir, 'temp.mp4')
      logging.info('Copying the transcoded file into place: %s -> %s' % ((video_name + ' (h264)' + '.mp4'), original_video_dir))
      shutil.copyfile(output_file, os.path.join(original_video_dir, (video_name + ' (h264)' + '.mp4') ) )
      logging.info('Deleting the original file: %s in %s' % (video_basename, original_video_dir))
      os.unlink(os.path.join(original_video_dir, video_basename))

it errors in plex with
___________________________________-

Sanity checking our work... 
2017-10-17 14:03:10,357 [b2de3b] Output file size looked sane, we'll replace the original: 428.0MB -> 428.0MB 
2017-10-17 14:03:10,357 [b2de3b] Copying the transcoded file into place: PAW Patrol (2013) - S03E25 - Pups Raise the Paw Patroller; Pups Save the Crows (h264).mp4 -> /Videos/Kid TV Shows/.grab/a79adac60673d91696793566ed5063cfb2acd9b9 
2017-10-17 14:03:10,361 [b2de3b] Something went wrong during sanity check: [Errno 2] No such file or directory: '/home/ryancox/tmp/b2de3b66-0bc9-4868-95ee-78e7a2c40865/temp.mp4' 
2017-10-17 14:03:10,361 [b2de3b] Leaving temp files in: /home/ryancox/tmp/b2de3b66-0bc9-4868-95ee-78e7a2c40865 
2017-10-17 14:03:10,361 [b2de3b] Done processing! 
________________________________
